apps/dashboard/app/controllers/projects_controller.rb

```diff
@@ -32,6 +35,11 @@ def show
   def index
     @projects = Project.all
     @templates = templates
+
+    if @@cache_shared_path
+      self.save_shared_dir
+      @@cache_shared_path = false
+    end
```

Comment:
IDK what this is, but you can use Rails.cache to cache things. You can grep through the codebase for examples.